Ensure go-basic and go has redundant links removed

```
Removed: 2187 links
 :: ERROR: multiply-labeled-edge: GO:0035278 GO:0006412 --> regulates negatively_regulates
 Makefile:188: recipe for target 'go-basic.obo' failed
make: *** [go-basic.obo] Error 1
Build step 'Execute shell' marked build as failure
```

These redundant links should be removed.

Here is the diagnosis:

We first create go-plus using the composition `reason o relax o reduce` in robot. This ensures a 'perfect' graph.

```
# reasoned.owl is equivalent to editors source, after reason-relax-reduce pipeline
# note: we keep this as a distinct intermediate target as the ontology IRI needs to be 'go' for Oort to work...
reasoned.owl: $(SRC) $(SRC)-check
	$(ROBOT)  reason -i $< -r ELK relax reduce -r ELK annotate -V $(RELEASE_URIBASE)/$(ONT).owl -o $@
```

As a next step we roll up object properties to the set of GO_MAIN_RELATIONS:

```
# basic relations are part_of & 3 regulations relations
GO_BASIC_RELATIONS = BFO:0000050 RO:0002211 RO:0002212 RO:0002213

# extend with has_part, occurs_in and during relations
# - this is the set used in the main go file
GO_MAIN_RELATIONS = $(GO_BASIC_RELATIONS) BFO:0000051 BFO:0000066 RO:0002091 RO:0002092 RO:0002093

...

	owltools $(USECAT) $< --remove-imports-declarations --remove-dangling --make-subset-by-properties $(GO_MAIN_RELATIONS) --set-ontology-id $(OBO)/go.owl -o -f obo $@.tmp && egrep -v '(property_value|relationship): never_in_taxon' $@.tmp > $@

```

The roll-up step does what it says on the tin and no more. 

as a side-effect it introduces **new** redundancies:

```
id: GO:0035278
name: miRNA mediated inhibition of translation
is_a: GO:0040033 ! negative regulation of translation, ncRNA-mediated
**relationship: negatively_regulates GO:0006412 ! translation
relationship: part_of GO:0035195 ! gene silencing by miRNA
**relationship: regulates GO:0006412 ! translation
```

The solution here is to run the reduce step *again*, post-roll-up. And then again for the *second* roll-up for go-basic

--- a/src/ontology/Makefile
+++ b/src/ontology/Makefile
@@ -162,12 +162,15 @@ $(GO_PLUS).owl: reasoned.owl
 #  - ...
 # go.obo is a simplified; we use oort to remove imports etc
 
-$(ONT)-pre.owl: reasoned.owl
+$(ONT)-pre.obo: reasoned.owl
 	owltools $(USECAT) $< --remove-imports-declarations --remove-dangling --make-subset-by-properties $(GO_MAIN_RELATIONS) --set-ontology-id $(OBO)/go.owl -o -f obo $@.tmp && egrep -v '(property_value|relationship): never_in_taxon' $@.tmp > $@
+
+$(ONT)-pre-reduced.obo: $(ONT)-pre.obo
+	robot reduce -r elk -i $< -o $@
 
 # use OORT
 # side effects: also builds go-simple.obo
-$(ONT).obo: $(ONT)-pre.owl
+$(ONT).obo: $(ONT)-pre-reduced.obo
 	ontology-release-runner --catalog-xml catalog-v001.xml --ignoreLock --skip-release-folder --skip-format owx --skip-format metadata --outdir $(BUILD_DIR) --allow-overwrite --asserted --simple --no-reasoner --remove-trailing-qualifiers $< &&\
 	$(PERLCHECK) --disable-xrf-abbs-check --disable-multiply-labeled-edge $(BUILD_DIR)/go.obo &&\
 	cp $(BUILD_DIR)/subsets/* subsets/ &&\
@@ -183,8 +186,11 @@ go.owl: $(ONT).obo
 go-simple-filtered.obo: go-simple.obo
 	owltools $< --make-subset-by-properties $(GO_BASIC_RELATIONS) // -o -f obo $@
 
+go-simple-filtered-reduced.obo: go-simple-filtered.obo
+	robot reduce -r elk -i $< -o $@
+
 # filter inter-ontology (e.g. MF to BP) links
-go-basic.obo: go-simple-filtered.obo
+go-basic.obo: go-simple-filtered-reduced.obo
 	$(PERLFILTER) $< > $@.tmp && mv $@.tmp $@ && $(PERLCHECK) --disable-xrf-abbs-check $@
 
 go-basic.owl: go-basic.obo

